dll/win32/browseui/internettoolbar.cpp

```diff
@@ -1930,3 +1986,19 @@ LRESULT CInternetToolbar::OnSettingsChange(UINT uMsg, WPARAM wParam, LPARAM lPar
 
     return 0;
 }
+
+HRESULT CInternetToolbar::GetStream(UINT StreamFor, DWORD Stgm, IStream **ppS)
+{
+    WCHAR path[MAX_PATH];
+    LPCWSTR subkey = NULL;
+    switch (StreamFor)
+    {
+    case ITBARSTREAM_SHELLBROWSER: subkey = L"ShellBrowser"; break;
+    case ITBARSTREAM_WEBBROWSER: subkey = L"WebBrowser"; break;
+    case ITBARSTREAM_EXPLORER: subkey = L"Explorer"; break;
+    default: return E_INVALIDARG;
+    }
+    wsprintfW(path, L"Software\\Microsoft\\Internet Explorer\\Toolbar\\%s", subkey);
```

Comment:
I'd prefer the StringCch family of functions.

Reply:
This string will **never** be longer than 54 characters.